metricbeat/module/system/process/helper.go

```diff
@@ -94,6 +95,11 @@ func (proc *Process) getDetails(envPredicate func(string) bool) error {
 	if err := proc.Cpu.Get(proc.Pid); err != nil {
 		return fmt.Errorf("error getting process cpu time for pid=%d: %v", proc.Pid, err)
 	}
+	
+	proc.IO = sigar.ProcIO{}
+	if err := proc.IO.Get(proc.Pid); err != nil {
+		return fmt.Errorf("error getting process io for pid=%d: %v", proc.Pid, err)
```

Comment:
You are going to need to do some error handling here. For example, on Windows where this is not implemented (yet?) this error will always occur and no process metrics will be reported.

FreeBSD shares the sigar_linux_common.go implementation. I don't know if the FreeBSD `linprocfs` exposes the `/proc/<pid>/io` so this could be another error that needs to be handled.